File: frontend/console/src/api/signal-si.js
import Axios from "@/http";

export function createSignalSi(params) {
    return new Promise((resolve, reject) => {
        Axios.post('/config/signal-si', params, { isJSON: true }).then(res => {
            resolve(res)
        })
    })
}

export function retrieveSignalSi(signalSiId) {
    return new Promise((resolve, reject) => {
        Axios.get(`/config/signal-si/${ signalSiId }`).then(res => {
            resolve(res)
        })
    })
}

export function updateSignalSi(params) {
    return new Promise((resolve, reject) => {
        Axios.put('/config/signal-si', params, { isJSON: true }).then(res => {
            resolve(res)
        })
    })
}

export function deleteSignalSi(signalSiId) {
    return new Promise((resolve, reject) => {
        Axios.delete(`/config/signal-si/${ signalSiId }`).then(res => {
            resolve(res)
        })
    })
}

export function deleteSignalSiWithFlag(signalSiId) {
    return new Promise((resolve, reject) => {
        Axios.put(`/config/signal-si/${ signalSiId }`).then(res => {
            resolve(res)
        })
    })
}

export function getAllSignalSis() {
    return new Promise((resolve, reject) => {
        Axios.get('/config/signal-si/all').then(res => {
            resolve(res)
        })
    })
}

export function querySignalSi (params) {
    return new Promise((resolve, reject) => {
        Axios.post('/config/signal-si/query', params, { isJSON: true }).then(res => {
            resolve(res)
        })
    })
}

File: little-bee/controller/router/signal_si.go
package router

import (
	"controller/config"

	"github.com/gin-gonic/gin"
)

const URL_ROUTER_SIGNAL_SI = "/config/signal-si"
const URL_ROUTER_SIGNAL_SI_BY_ID = "/config/signal-si/:signal-si-id"
const URL_ROUTER_SIGNAL_SI_QUERY = "/config/signal-si/query"
const URL_ROUTER_SIGNAL_SI_ALL = "/config/signal-si/all"

func SignalSiRouter(r *gin.RouterGroup) *gin.RouterGroup {
	r.POST(URL_ROUTER_SIGNAL_SI, config.CreateSignalSi)
	r.GET(URL_ROUTER_SIGNAL_SI_BY_ID, config.RetrieveSignalSi)
	r.PUT(URL_ROUTER_SIGNAL_SI, config.UpdateSignalSi)
	r.DELETE(URL_ROUTER_SIGNAL_SI_BY_ID, config.DeleteSignalSi)
	r.POST(URL_ROUTER_SIGNAL_SI_QUERY, config.QuerySignalSi)
	r.GET(URL_ROUTER_SIGNAL_SI_ALL, config.GetAllSignalSis)
	return r
}